### PR TITLE
Handle auth errors for entity detail fetches

### PR DESCRIPTION
**Pull request by fomentia**

### CODE REVIEW

File: src/okta/fetchBatchOfApplicationUsers.ts [hunk 1]
```diff
@@ -100,18 +100,17 @@ export default async function fetchBatchOfApplicationUsers(
 
       // Create application user iterator for current applicationIndex, picking
       // up at the page represented by the last obtained pagination token.
-      let listApplicationUsers: OktaCollection<OktaApplicationUser>;
-      try {
-        listApplicationUsers = await okta.listApplicationUsers(
```
Comment (fomentia) [inline src/okta/fetchBatchOfApplicationUsers.ts]: This method returns a Collection object, which doesn't make any actual API calls until you start iteration. This old code was doing nothing. :expressionless:

**Comment by fomentia** [comment]
It turned out to be an API call we were making to get details for an entity, not the listing API call.

File: src/okta/fetchBatchOfApplicationUsers.ts (hunk 1 above)
Comment (aiwilliams) [reply src/okta/fetchBatchOfApplicationUsers.ts]: Is there any way to add an error handler to the collection before iterating, or perhaps provide one to the iterator function?

Comment (fomentia) [reply src/okta/fetchBatchOfApplicationUsers.ts]: Sadly no. This is the interface of the Collection object: https://github.com/okta/okta-sdk-nodejs/blob/master/src/collection.js

**Pull request merged**